### PR TITLE
Reuse HTTP connections.

### LINKED ISSUE #2778: Connectionpool not pooling
When I run with `-v`, I see entries like this repeatedly:

2016-04-07 05:23:55,283:INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
2016-04-07 05:23:58,509:INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org

This probably means that the connectionpool is not actually reusing connections properly, which makes the process slower than it should be.

### PR DESCRIPTION
**Pull request by jsha**
Fixes #2778

### CODE REVIEW

**Comment by joohoi** [comment]
IIRC requests.session() leaves the open connection hanging and its use in classes is best accompanied by self.session.close() in magic method __del__

**Comment by jsha** [comment]
I fixed the tests and this is now ready for review again.

File: acme/acme/client_test.py [hunk 1]
```diff
@@ -528,40 +528,48 @@ def test_check_response_jobj(self):
...
+        self.net._send_request('GET', 'http://example.com/',
+                               headers={'User-Agent': 'foo2'})
+        self.net.session.request.assert_called_with(
+            'GET', 'http://example.com/',
+            verify=mock.ANY, headers={'User-Agent': 'foo2'})
+
+    def test_del(self):
+        self.net.session = mock.MagicMock()
+        self.net.__del__()
+        self.net.session.close.assert_called_once()
```
Comment (joohoi) [inline acme/acme/client_test.py]: something along the lines of:
```
          sess = mock.MagicMock()
          self.net.session = sess
          del self.net
          sess.close.assert_called_once()
```
would be more "true" to the actual functionality.

**Comment by joohoi** [comment]
LGTM despite the nit above

**Comment by jsha** [comment]
Fixed the nit.

**Pull request merged**